src/mesh/mesh_triangle_interface.C

```diff
@@ -106,13 +106,18 @@ void TriangleInterface::triangulate()
   // triangulate is now sitting in the mesh.
 
   // If the holes vector is non-nullptr (and non-empty) we need to determine
-  // the number of additional points which the holes will add to the
-  // triangulation.
+  // the number of additional points and segments which the holes will add to
+  // the triangulation.
   unsigned int n_hole_points = 0;
+  unsigned int num_hole_segments = 0;
 
   if (have_holes)
     for (const auto & hole : *_holes)
+    {
       n_hole_points += hole->n_points();
+      for (unsigned int i=0; i<hole->segment_indices().size()-1; ++i)
+        num_hole_segments += hole->segment_indices()[i+1] - hole->segment_indices()[i];
+    }
```

Comment:
It appears that `segment_indices()` function always returns a vector with exactly 2 entries? And even if it didn't this loop would add/subtract the intermediate values anyway so you would only end up with the difference between the first and last entry. Not saying this is wrong, just trying to understand what's going on.

Reply:
Yes, the size of `segment_indices()` is `n`+1, where `n` is the number of segments described by nodes on the segments.

Reply:
For example of a hole created by two concentric circles in the ring. In such a case, `n` will be 2. The entries in `segment_indices()` will be 3. The nodes are ordered properly so that we can have implicit rule on adding segments with just these three numbers.

Reply:
I will add an assertion to make sure we have a peaceful mind ;-)

Reply:
What I'm saying is, if segment_indices = [a,b,c,d,..., z] then the result of your calculation above is simply `num_hole_segments` = z-a because all the intermediate terms cancel. So there's no reason to do this loop?

Reply:
Yes, I think we just need to fix Line 138.